### PR TITLE
Update resource types to use hardened images

### LINKED ISSUE #34: Replace docker resource types with hardened images for pipelines (mailer)
To improve our security and compliance posture, we want to update our CI resource types to use cloud.gov's hardened image resources. Look at [common-pipelines](https://github.com/cloud-gov/common-pipelines) to determine all of the currently hardened resource images. Some possible images are:

- cron-resource
- general-task
- github-pr-resource
- s3-resource
- s3-simple-resource
- slack-notification-resource
- cf-cli-resource 
- cf-resource 
- email-resource
- git-resource 
- registry-image-resource 
- semver-resource 
- time-resource

## Acceptance Criteria

- [ ] Updates all relevant CI resource types
- [ ] Verify CI jobs and tasks run as expected
- [ ] Document the updated resources types

### PR DESCRIPTION
**Pull request by svenaas**
Fixes #34 

## Changes proposed in this pull request:
- Update pipelines to use our current hardened images

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations
Replaces resources with hardened images

### CODE REVIEW

**Review by apburnes — CHANGES_REQUESTED** [review]
We'll want to use our `pages-node-v20` for the node image and update the package.json, add the git and time hardened resource types, and use the `harden-concourse-task` for the cf-image.

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@fastify/basic-auth": "^5.0.0",
     "fastify": "^4.21.0",
     "fastify-plugin": "^4.5.1",
+    "node": "^20.11.0",
```
Comment (apburnes) [inline package.json:27]: We'll want to update the `engines` key in the package and `npm uninstall` this `node` from the dependencies.

The engines should look like:
```json
"engines": {
  "node": "^20.x.x"
}
```

File: ci/pipeline.yml [hunk 2]
```diff
@@ -11,17 +11,26 @@ email-cf: &email-cf
   CF_STACK: cflinuxfs4
 
 node-image: &node-image
```
Comment (apburnes) [inline ci/pipeline.yml:13]: This happened to me while updating the redirects CI task. 

You'll want to configure it to look like:
```yml
  config:
    <<: *node-image
```

and not 
```yml
  config:
    platform: linux
    image_resource:
      <<: *node-image
```

We added the `platform` and `image_resource` keys to the `node-image` anchor so we don't need to call them again when using it in a task.

File: package.json (hunk 1 above)
Comment (svenaas) [reply package.json]: Hmm, OK. How's this?

**Review by apburnes — COMMENTED** [review]
I noticed this repo uses a `.nvmrc` file with node v18. We can just delete that file because the package.json engine defines the node version. After that we should be good

**Comment by svenaas** [comment]
.nvmrc removed. Re-review requested. Should I try manually flying this before merging?

**Review by apburnes — APPROVED** [review]
Good to go 🚀

**Pull request merged**